src/rpc/core_rpc_server_commands_defs.h

```diff
@@ -1964,6 +1964,28 @@ namespace rpc {
     };
   };
 
+  OXEN_RPC_DOC_INTROSPECT
+  // Accesses the amounts accrued to addresses in the batching database
+  struct GET_ACCRUED_BATCHED_EARNINGS: PUBLIC
+  {
+    static constexpr auto names() { return NAMES("get_accrued_batched_earnings"); }
+
+    struct request
+    {
+      std::vector<std::string> addresses; // Array of addresses to query the batching database about.
+
+      KV_MAP_SERIALIZABLE
+    };
+
+    struct response
+    {
+      std::string status;            // Generic RPC error code. "OK" is the success value.
+      std::vector<uint64_t> amounts; // An array of amounts according to the provided addressses
```

Comment:
This is going to require a wallet to reveal its wallet address to public service nodes, which we want to avoid.  Instead I think we need this endpoint to accept *no* addresses and return a big dict of:
```json
{
    "a1": 12345,
    "a2": 45678,
    "a3": 5017012345
}
```
which contains *all* owed balances.  If you *do* specify an address, then we only include the ones you specify.

Reply:
But epee serialization is garbage and it doesn't look to me like it's even possible to serialize that.

I'm tempted to bring in the nlohmann::json dependency early for this endpoint just to make it not garbage.

Reply:
I think we push the change after bt-rpc gets merged. Then make the address param optional which will return the entire database then.

We have to redo this function anyway to suit bt-rpc. No use refactoring it twice